Makefile

```diff
@@ -31,40 +32,59 @@ all: docker
 gobuild:
 	./scripts/build false
 
+# create output directories
+.out-stamp:
+	mkdir -p ./out/test-artifacts ./out/cni-plugins
+	touch .out-stamp
+
+out: .out-stamp
```

Comment:
Seems like `out` is a nop, why is it needed?

Reply:
I want to ensure that the `./out/*` is created by the user running `make`.  If docker creates a folder structure, it will be owned by root and it breaks `make clean`.

I'm putting this here so that it can be reused by the multiple targets that need it.

Reply:
But, that's done in the `.out-stamp` target. Isn't it? `out` itself isn't doing anything apart from depending on `.out-stamp`. Is there some obvious `Makefile` logic in play here that I'm missing?

Reply:
Oh right right, I see what you're asking.

The make target needs to be the same name as the output, which is why .out-stamp creates .out-stamp. However, `.out-stamp` doesn't convey much to the targets that depend on it, so we use `out` to have a more meaningful name.

I'm not just using `out` because I read some advice online that suggets not using directories as makefile targets.